BGL/include/CGAL/boost/graph/helpers.h

```diff
@@ -514,134 +536,159 @@ return valid;
+
+  std::size_t hn = 0;
+  BOOST_FOREACH(halfedge_descriptor i, halfedges(g))
+  {
+    ++hn;
+
     //counting borders
-    if ( is_border(i, g))
+    if(is_border(i, g))
       ++nb;
+
     // face integrity.
-    
-    valid = valid && ( face(i, g) == face(next(i, g), g));
-    if ( ! valid) {
-      verr << "    face pointer integrity2 corrupted."
-           << std::endl;
+    valid = valid && (face(i, g) == face(next(i, g), g));
+    if(!valid)
+    {
+      verr << "halfedge " << hn << "    face(hd) != face(next(hd))." << std::endl;
       break;
     }
   }
+
   verr << "sum border halfedges (2*nb) = " << 2 * nb << std::endl;
-  if ( valid && n + nb  != num_h)
+  if(valid && n + nb != num_h)
     verr << "counting halfedges via faces failed." << std::endl;
-  valid = valid && ( f == num_f);
-  valid = valid && ( n + nb  == num_h);
-  verr << "is_valid(): structure is " << ( valid ? "valid." :
-                                                   "NOT VALID.") << std::endl;
+
+  valid = valid && (f == num_f);
+  valid = valid && (n + nb  == num_h);
+  verr << "Face Graph Structure is " << (valid ? "valid." : "NOT VALID.") << std::endl;
+
   return valid;
 }
+
 /*!
   \ingroup PkgBGLHelperFct
  * \brief checks the integrity of `g`.
- * 
+ *
  * `g` is valid if it is a valid `FaceListGraph` and it has distinct faces on each side of an edge.
  * calls `is_valid_face_graph()`.
- * 
+ *
  * \param g the `Mesh` to test.
  * \param verb : if `true`, the details of the check will be written in the standard output.
  *
- * \tparam Mesh a model of `FaceListGraph` and `HalfedgeListGraph`, and follows 
+ * \tparam Mesh a model of `FaceListGraph` and `HalfedgeListGraph`, and follows
  * the definition of a \ref PMPDef "PolygonMesh"
  * \return `true` if `g` is valid, `false` otherwise.
- * 
+ *
  * \see `is_valid_face_graph()`
  * \see `is_valid_halfedge_graph()`
- * 
+ *
  */
 template <typename Mesh>
 bool is_valid_polygon_mesh(const Mesh& g, bool verb = false)
 {
   typedef typename boost::graph_traits<Mesh>::halfedge_descriptor   halfedge_descriptor;
+
   Verbose_ostream verr(verb);
- bool valid=is_valid_face_graph(g, verb);
- //test for 2-manifoldness
- // Distinct facets on each side of an halfedge.
- BOOST_FOREACH(halfedge_descriptor i, halfedges(g)){
-   valid = valid && (face(i, g) != face(opposite(i, g), g));
-   if ( ! valid) {
-     verr << "    both incident facets are equal." << std::endl;
-     break;
-   }
-   valid = valid && ( next(next(i, g), g) != i);
-   valid = valid && ( target(i, g) != target(next(i, g), g));
-   valid = valid && ( target(i, g) != target(next(next(i, g), g), g));
-   if ( ! valid) {
-     verr << "    incident facet is not at least a triangle."
-          << std::endl;
-     break;
-   }
-   if ( ! valid) {
-       verr << "    incident facet is not at least a triangle."
-            << std::endl;
-       break;
-   }
- }
- return valid;
-}
+  bool valid = is_valid_face_graph(g, verb);
+  if(!valid)
```

Comment:
Is this the only fix in the patch? putting all in the same commit makes it hard to spot the real fixes.

Reply:
https://github.com/CGAL/cgal/pull/3858/files?utf8=%E2%9C%93&diff=split&w=1 and it's pretty clear. What you're quoting is just an early exit. What was broken is not having a break in loops such as [here](https://github.com/CGAL/cgal/pull/3858/files#diff-dbc8316e37114ea33573d304bd560faaR482).

Reply:
That's what I did and I can't agree. Having two commits, one with the extra breaks and one with the cosmetic changes would have made the change obvious. Thanks for pointing the line.